internal/mage/sdk/elixir.go

```diff
@@ -156,16 +155,11 @@ func (Elixir) Publish(ctx context.Context, tag string) error {
 	defer c.Close()
 
 	var (
-		version   = strings.TrimPrefix(tag, "sdk/elixir/v")
-		mixFile   = "sdk/elixir/mix.exs"
-		hexAPIKey = os.Getenv("HEX_API_KEY")
-		dryRun    = os.Getenv("HEX_DRY_RUN")
+		version = strings.TrimPrefix(tag, "sdk/elixir/v")
+		mixFile = "sdk/elixir/mix.exs"
+		dryRun  = os.Getenv("HEX_DRY_RUN")
 	)
 
-	if hexAPIKey == "" {
-		return errors.New("HEX_API_KEY environment variable must be set")
```

Comment:
This guard was here so that if this was run without the necessary environment variable, it would fail with a clear error message. I think that we should keep it, but it's also OK if we want to remove it. If that's the case, then we should make the same change for the Python SDK too.

Reply:
We switch this to `util.HostSecretVar`: https://github.com/dagger/dagger/blob/79643efbae9ede709a28a52b9462f756a3654238/internal/mage/util/util.go#L186-L201

This has this check already.